Source/WebCore/platform/mediastream/RealtimeVideoCaptureSource.cpp

```diff
@@ -225,9 +222,21 @@ bool RealtimeVideoCaptureSource::supportsCaptureSize(std::optional<int> width, s
     return false;
 }
 
-bool RealtimeVideoCaptureSource::shouldUsePreset(const VideoPreset& current, const VideoPreset& candidate)
+static bool shouldUsePreset(const VideoPreset& current, const VideoPreset& candidate, bool shouldPreferPowerEfficiency)
 {
-    return candidate.size().width() <= current.size().width() && candidate.size().height() <= current.size().height() && prefersPreset(candidate);
+    if (shouldPreferPowerEfficiency && candidate.isEfficient() && !current.isEfficient())
+        return true;
```

Comment:
If more than one preset is power efficient, should we prefer the one that is closest to the desired size?

Reply:
I think this is what next line will do